rpc/types/utils.go

```diff
@@ -292,6 +287,41 @@ func BaseFeeFromEvents(events []abci.Event) *big.Int {
 	return nil
 }
 
+// EffectiveGasPrice computes the transaction gas fee, based on the given baseFee value.
+//
+// For EIP-1559 transactions:
+//
+//	price = min(gasTipCap + baseFee, gasFeeCap)
+//
+// For legacy transactions:
+//
+//	price = gasPrice
+//
+// This method is based on go-ethereum's internal effectiveGasPrice calculation.
+// (https://github.com/ethereum/go-ethereum/blob/d818a9af7bd5919808df78f31580f59382c53150/internal/ethapi/api.go#L1083-L1093)
+func EffectiveGasPrice(tx *ethtypes.Transaction, baseFee *big.Int) *big.Int {
+	if tx == nil {
+		return big.NewInt(0)
+	}
+
+	// For legacy and access list transactions, return the gas price directly
+	if tx.Type() != ethtypes.DynamicFeeTxType {
+		return tx.GasPrice()
```

Comment:
Q: what about tx type `Blob` and `SetCode`, can EIP-1559 apply on them? Can they also have non-nil `baseFee`?